src/lexer/token.py

```diff
@@ -5,14 +5,21 @@
 class TokenType(Enum):
     def __init__(self, token: str, delim_id: str):
         self._token = token
-        self.delim_id = delim_id
-
-        self.expected_delims = DELIMS[self.delim_id]
+        self._delim_id = delim_id
+        self._expected_delims = DELIMS[self.delim_id]
 
     @property
     def token(self):
         return self._token
 
+    @property
+    def delim_id(self):
+        return self._delim_id
+
+    @property
+    def expected_delims(self):
+        return self._expected_delims
+
     def __str__(self):
         return self.token
```

Comment:
if we're going to keep this, do we still need:
```
@property
def token(self):
   return self._token
```
? seems redundant

Reply:
only reason why i put the `__str__` is because magkakaron tayo token.token.token reference dito and that bothered me HAHAHAHAH

```
    for token in x.tokens:
        print(f"{token.lexeme}", end="\t")
        print(f"{token.token}", end="\t") # <<< right here!!
        print(f"{token.position} - {token.end_position}", end="\t")
        print()
```

It should be fine without it, though. Just gotta change how we reference it